server/src/controllers/baseLoad.ts

```diff
@@ -9,22 +9,50 @@ import { getDefaultBaseData } from "../data/getDefaultBaseData";
 import { FilterFrontendKeys } from "../utils/FrontendKey";
 import { flags } from "../data/flags";
 import { getCurrentDateTime } from "../utils/getCurrentDateTime";
+import {WorldMapCell} from "../models/worldmapcell.model";
 
+
+interface BaseLoadRequest {
+  type: string
+  userid: string
+  baseid: string
+  cellid: string
+}
 // MR2 ToDo: The client sends this data to the server: {"baseid":"1234","type":"view","userid":""}
 // In this example, the baseid '1234' is a hardcoded value in wildMonsterCell.ts for a tribe's base,
 // The 'baseid' should be used to lookup & return a base in the database with the corresponding id
 export const baseLoad: KoaController = async (ctx) => {
+  const requestBody: BaseLoadRequest = <BaseLoadRequest>ctx.request.body
+  console.log(requestBody.baseid)
   // Try find an already existing save
   const user: User = ctx.authUser;
   await ORMContext.em.populate(user, ["save"]);
 
+
   let save = user.save;
+
+
+
+  if (requestBody.cellid && requestBody.baseid) {
+    const cell = await ORMContext.em.findOne(WorldMapCell, {
+      cell_id: parseInt(requestBody.cellid),
+    })
+    if (cell) {
+      if (cell.base_type) {
+        save = await ORMContext.em.findOne(Save, {
+          basesaveid: cell.base_id,
+        })
+      }
+    }
+  }
+
```

Comment:
Redundant nesting for if conditions here.

```javascript 
if (cell && cell.base_type)
```

Reply:
this is actually incomplete, it should really check for the base_type because our plan is to use different table for the save, tribes, outposts right?

Reply:
Yeah exactly